### PR TITLE
Logging, simplified

### PR DESCRIPTION
**Pull request by gcolvin**
Simplified by removing unused - and likely unneeded - functionality.  So just one logger, and easier to reset streams logged to.

Also wrapped logging in a mutex so that the logs of multiple threads interleave cleanly.

### CODE REVIEW

File: db/silkworm/common/log.hpp [hunk 1]
```diff
@@ -17,48 +17,49 @@
...
+//
+#define SILKWORM_LOG_STREAMS(stream1_, stream2_) log_set_streams_((stream1_), (stream2_));
+
+// silence
+std::ostream& null_stream();
+
+//
+// Below are for access via macros ONLY :(
+//
+LogLevels log_verbosity_();
```
Comment (yperbasis) [inline db/silkworm/common/log.hpp:47]: I'd wrap stuff that's not part of the API into `detail` namespace (like they do in Boost).

**Review by yperbasis — APPROVED** [review]

File: db/silkworm/common/log.hpp (hunk 1 above)
Comment (gcolvin) [reply db/silkworm/common/log.hpp:47]: Good idea.  Looks like this will our first use of that namespace.

**Review by AndreaLanfranchi — COMMENTED** [review]
Sorry for being pedantic.
When we touch a file we should update copyright year (don't kill me)

File: db/silkworm/common/log.hpp (hunk 1 above)
Comment (yperbasis) [reply db/silkworm/common/log.hpp:47]: It's used in `db/silkworm/db/util.hpp` already :)

**Pull request merged**